packages/owlbot-bootstrapper/common-container/github-authenticator.ts

```diff
@@ -30,16 +30,19 @@ export class GithubAuthenticator {
   projectId: string;
   appInstallationId: string;
   secretManagerClient: SecretManagerServiceClient;
+  signingClient: typeof sign;
   OWLBOT_SECRET_NAME = 'owlbot-bootstrapper';
 
   constructor(
     projectId: string,
     appInstallationId: string,
-    secretManagerClient: SecretManagerServiceClient
+    secretManagerClient: SecretManagerServiceClient,
+    signingClient: typeof sign
```

Comment:
I would declare your own type for this rather than having it dynamically typed based on a third-party library. You could define this type as `(payload: JwtPayload, privateKey: string, options: SomeOptionInterface) => string` which jwt's `sign` would match. If the third-party interface breaks, then your interface breaks. You are also exposing more options than necessary to the public interface.

Another option is to declare this as a simpler type like `(payload: object, privateKey: string) => string` and have your default implementation:

```ts
function jwtSigner(payload: object, privateKey: string): string {
  return sign(payload, privateKey, {algorithm: 'RS256'});
}
```